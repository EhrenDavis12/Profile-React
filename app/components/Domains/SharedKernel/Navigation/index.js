/**
 *
 * Navigation
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';
import { Link } from 'react-router-dom';
import './index.css';

function Navigation() {
  const nav = (
    <nav>
      <ul>
        <li>
          <Link to="/">About Me</Link>
        </li>
        <li>
          <Link to="/portfolio">Portfolio</Link>
        </li>
        <li>
          <Link to="/contact">Contact</Link>
        </li>
        <li>
          <Link to="/messages">Messages</Link>
        </li>
      </ul>
    </nav>
  );
  return nav;
}

Navigation.propTypes = {};

export default Navigation;
