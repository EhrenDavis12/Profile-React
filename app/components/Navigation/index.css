ul.unstyled {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

nav ul {
  opacity: 0.8;
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  background-image: linear-gradient(
    to bottom right,
    rgb(8, 253, 253),
    rgb(0, 26, 255)
  );
}

nav li {
  float: left;
}

nav li a {
  display: block;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

nav li a:hover {
  background-color: rgb(8, 253, 253);
  color: white;
}

nav button {
  margin-top: 13px;
}
