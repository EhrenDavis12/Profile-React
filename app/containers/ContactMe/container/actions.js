/*
 *
 * ContactMePage actions
 *
 */

import {
  SUBMIT_MESSAGE_FORM,
  SUBMIT_MESSAGE_FORM_SUCCEEDED,
  SUBMIT_MESSAGE_FORM_FAILED,
  CLOSED_SUCCESS_MESSAGE,
} from './constants';

export function submitMessageForm(messageForm) {
  return {
    type: SUBMIT_MESSAGE_FORM,
    messageForm,
  };
}

export function submitMessageFormSucceeded() {
  return {
    type: SUBMIT_MESSAGE_FORM_SUCCEEDED,
  };
}

export function submitMessageFormFailed(message) {
  return {
    type: SUBMIT_MESSAGE_FORM_FAILED,
    message,
  };
}

export function closedSuccessMessage() {
  return {
    type: CLOSED_SUCCESS_MESSAGE,
  };
}
