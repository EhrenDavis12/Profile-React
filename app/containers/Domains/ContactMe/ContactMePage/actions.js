/*
 *
 * ContactMePage actions
 *
 */

import {
  SUBMIT_MESSAGE_FORM,
  SUBMIT_MESSAGE_FORM_SUCCEEDED,
  SUBMIT_MESSAGE_FORM_FAILED,
} from './constants';

export function submitMessageForm(messageForm) {
  return {
    type: SUBMIT_MESSAGE_FORM,
    messageForm,
  };
}

export function submitMessageFormSucceeded() {
  return {
    type: SUBMIT_MESSAGE_FORM_SUCCEEDED,
  };
}

export function submitMessageFormFailed(message) {
  return {
    type: SUBMIT_MESSAGE_FORM_FAILED,
    message,
  };
}
