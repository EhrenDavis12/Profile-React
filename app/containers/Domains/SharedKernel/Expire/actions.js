/*
 *
 * Expire actions
 *
 */

import {
  CLEAR_TIMER_ACTION,
  SET_TIMER_ACTION,
  SHOW_CHILDREN_ACTION,
  HIDE_CHILDREN_ACTION,
  SET_TIMER_SUCCEEDED,
  CLEAR_TIMER_SUCCEEDED,
} from './constants';

export function setTimerAction(
  delay,
  onHideChildrenAction,
  onClearTimerAction,
) {
  return {
    type: SET_TIMER_ACTION,
    delay,
    onHideChildrenAction,
    onClearTimerAction,
  };
}

export function setTimerSucceeded(timer) {
  return {
    type: SET_TIMER_SUCCEEDED,
    timer,
  };
}

export function clearTimerAction() {
  return {
    type: CLEAR_TIMER_ACTION,
  };
}

export function clearTimerSucceeded() {
  return {
    type: CLEAR_TIMER_SUCCEEDED,
  };
}

export function showChildrenAction() {
  return {
    type: SHOW_CHILDREN_ACTION,
  };
}

export function hideChildrenAction() {
  return {
    type: HIDE_CHILDREN_ACTION,
  };
}
