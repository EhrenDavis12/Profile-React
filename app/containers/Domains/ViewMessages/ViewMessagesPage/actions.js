/*
 *
 * ViewMessagesPage actions
 *
 */

import {
  REQUEST_USER,
  REQUEST_USER_SUCCEEDED,
  REQUEST_USER_FAILED,
  REQUEST_USER_MESSAGES,
  REQUEST_USER_MESSAGES_SUCCEEDED,
  REQUEST_USER_MESSAGES_FAILED,
  SELECT_USER_MESSAGE,
} from './constants';

export function requestUser() {
  return {
    type: REQUEST_USER,
  };
}

export function requestUserSucceeded(user) {
  return {
    type: REQUEST_USER_SUCCEEDED,
    user,
  };
}

export function requestUserFailed(message) {
  return {
    type: REQUEST_USER_FAILED,
    message,
  };
}

export function requestUserMessages() {
  return {
    type: REQUEST_USER_MESSAGES,
  };
}

export function requestUserMessagesSucceeded(userMessages) {
  return {
    type: REQUEST_USER_MESSAGES_SUCCEEDED,
    userMessages,
  };
}

export function requestUserMessagesFailed(message) {
  return {
    type: REQUEST_USER_MESSAGES_FAILED,
    message,
  };
}

export function selectUserMessage(selectedUserMessage) {
  return {
    type: SELECT_USER_MESSAGE,
    selectedUserMessage,
  };
}
