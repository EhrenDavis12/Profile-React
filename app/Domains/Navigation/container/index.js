/**
 *
 * Navigation
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import NavigationBody from 'Domains/Navigation/components/NavigationBody';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import makeSelectNavigation from './selectors';
import reducer from './reducer';
import saga from './saga';
import { toggleDrawer, selectLink } from './actions';

/* eslint-disable react/prefer-stateless-function */
export class Navigation extends React.Component {
  render() {
    return (
      <div>
        <NavigationBody {...this.props} />
      </div>
    );
  }
}

Navigation.propTypes = {
  // dispatch: PropTypes.func.isRequired,
};

/* const mapStateToProps = createStructuredSelector({
  Navigation: makeSelectNavigation(),
}); */
const mapStateToProps = makeSelectNavigation();

export function mapDispatchToProps(dispatch) {
  return {
    toggleDrawer: () => dispatch(toggleDrawer()),
    selectItem: linkSelected => dispatch(selectLink(linkSelected)),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'Navigation', reducer });
const withSaga = injectSaga({ key: 'Navigation', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(Navigation);
