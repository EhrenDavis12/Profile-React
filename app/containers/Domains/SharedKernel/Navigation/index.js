/**
 *
 * Navigation
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

// import NavigationComponents from 'components/Domains/SharedKernel/Navigation';
import AppBar from 'components/Domains/SharedKernel/AppBar';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import makeSelectNavigation from './selectors';
import reducer from './reducer';
import saga from './saga';
import { toggleDrawer } from './actions';

/* eslint-disable react/prefer-stateless-function */
export class Navigation extends React.Component {
  render() {
    return (
      <div>
        <AppBar {...this.props} />
      </div>
    );
  }
}

Navigation.propTypes = {
  // dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  navigation: makeSelectNavigation(),
});
// const mapStateToProps = createStructuredSelector();

function mapDispatchToProps(dispatch) {
  return {
    toggleDrawer: () => dispatch(toggleDrawer()),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'navigation', reducer });
const withSaga = injectSaga({ key: 'navigation', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(Navigation);
