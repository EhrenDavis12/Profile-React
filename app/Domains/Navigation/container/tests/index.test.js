import React from 'react';
import 'jest-styled-components';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
// import { enzymeFind } from 'styled-components/test-utils';
import { browserHistory } from 'react-router-dom';
import configureStore from 'containers/../configureStore';
import { toggleDrawer, selectLink } from '../actions';

import Navigation, { mapDispatchToProps } from '../index';

describe('<Navigation />', () => {
  let store;
  let wrapper;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
    wrapper = renderComponentMount();
  });

  const renderComponentMount = () =>
    mount(
      <Provider store={store}>
        <Navigation />
      </Provider>,
    );

  describe('<Navigation /> snap Shots', () => {
    it('renders as expected with populated messages', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Navigation />', () => {
    it('should render its NavigationBody', () => {
      expect(wrapper.find('NavigationBody').length).toEqual(1);
    });
  });

  describe('mapDispatchToProps', () => {
    describe('toggleDrawer', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.toggleDrawer).toBeDefined();
      });

      it('should dispatch toggleDrawer when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.toggleDrawer();
        expect(dispatch).toHaveBeenCalledWith(toggleDrawer());
      });
    });

    describe('selectLink', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.selectItem).toBeDefined();
      });

      it('should dispatch selectLink when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const link = 'mxstbr';
        result.selectItem(link);
        expect(dispatch).toHaveBeenCalledWith(selectLink(link));
      });
    });
  });
});
