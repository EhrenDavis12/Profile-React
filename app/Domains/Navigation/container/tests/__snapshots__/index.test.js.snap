// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> <Navigation /> snap Shots renders as expected with populated messages 1`] = `
.c1 {
  padding: 25px;
  text-transform: uppercase;
  cursor: pointer;
  border-bottom: solid 1px #f6f6f6;
}

.c1:hover {
  background-color: #f6f6f6;
}

.c0 {
  padding-left: 0px;
  z-index: 900;
}

.c2 {
  position: fixed;
  top: 0;
  left: -300px;
  height: 100%;
  width: 300px;
  background: #fff;
  padding: 0px;
  -webkit-transition: 0.25s ease-in-out;
  transition: 0.25s ease-in-out;
  box-shadow: rgba(0,0,0,0.15) 0px 3px 10px,rgba(0,0,0,0.22) 0px 3px 10px;
  z-index: 1000;
  color: #000000;
}

.c2.drawerOpen {
  left: 0;
}

.c3 {
  padding: 25px;
  text-transform: uppercase;
  cursor: pointer;
  border-bottom: solid 1px #f6f6f6;
  display: block;
}

.c3:hover {
  background-color: #f6f6f6;
}

<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "Navigation": [Function],
      },
      "injectedSagas": Object {
        "Navigation": Object {
          "mode": "@@saga-injector/restart-on-remount",
          "saga": [Function],
          "task": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 1,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": null,
            "name": "rootSaga",
            "result": [Function],
            "setContext": [Function],
          },
        },
      },
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <withReducer(withSaga(Connect(Navigation)))>
    <withSaga(Connect(Navigation))>
      <Connect(Navigation)>
        <Navigation
          isDrawerOpen={false}
          navigationLinks={
            Array [
              Object {
                "DisplayName": "About Me",
                "LinkTo": "ProfessionalIntro",
                "key": "AboutMe",
              },
              Object {
                "DisplayName": "Hobbies",
                "LinkTo": "Hobbies",
                "key": "Hobbies",
              },
              Object {
                "DisplayName": "ContactMe",
                "LinkTo": "ContactMePage",
                "key": "ContactMe",
              },
            ]
          }
          selectItem={[Function]}
          toggleDrawer={[Function]}
        >
          <div>
            <NavigationBody
              isDrawerOpen={false}
              navigationLinks={
                Array [
                  Object {
                    "DisplayName": "About Me",
                    "LinkTo": "ProfessionalIntro",
                    "key": "AboutMe",
                  },
                  Object {
                    "DisplayName": "Hobbies",
                    "LinkTo": "Hobbies",
                    "key": "Hobbies",
                  },
                  Object {
                    "DisplayName": "ContactMe",
                    "LinkTo": "ContactMePage",
                    "key": "ContactMe",
                  },
                ]
              }
              selectItem={[Function]}
              toggleDrawer={[Function]}
            >
              <NavTopButtons
                itemKeyAttr="key"
                itemLabelAttr="DisplayName"
                items={
                  Array [
                    Object {
                      "DisplayName": "About Me",
                      "LinkTo": "ProfessionalIntro",
                      "key": "AboutMe",
                    },
                    Object {
                      "DisplayName": "Hobbies",
                      "LinkTo": "Hobbies",
                      "key": "Hobbies",
                    },
                    Object {
                      "DisplayName": "ContactMe",
                      "LinkTo": "ContactMePage",
                      "key": "ContactMe",
                    },
                  ]
                }
                selectItem={[Function]}
                toggleDrawer={[Function]}
              >
                <styles__Nav
                  className="navbar sticky-top navbar-expand-md navbar-light"
                >
                  <StyledComponent
                    className="navbar sticky-top navbar-expand-md navbar-light"
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "styles__Nav-sc-1iujtvx-3",
                          "isStatic": true,
                          "lastClassName": "c0",
                          "rules": Array [
                            "padding-left:0px;z-index:900;",
                          ],
                        },
                        "displayName": "styles__Nav",
                        "render": [Function],
                        "styledComponentId": "styles__Nav-sc-1iujtvx-3",
                        "target": "nav",
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <nav
                      className="navbar sticky-top navbar-expand-md navbar-light c0"
                    >
                      <styles__HamburgerButtonStyled
                        aria-controls="navbarNav"
                        aria-expanded="false"
                        aria-label="Toggle navigation"
                        className="navbar-toggler"
                        data-target="#navbarNav"
                        data-toggle="collapse"
                        onClick={[Function]}
                        type="button"
                      >
                        <StyledComponent
                          aria-controls="navbarNav"
                          aria-expanded="false"
                          aria-label="Toggle navigation"
                          className="navbar-toggler"
                          data-target="#navbarNav"
                          data-toggle="collapse"
                          forwardedClass={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": undefined,
                              "componentStyle": ComponentStyle {
                                "componentId": "styles__HamburgerButtonStyled-sc-1iujtvx-2",
                                "isStatic": true,
                                "lastClassName": "dqgFGn",
                                "rules": Array [],
                              },
                              "displayName": "styles__HamburgerButtonStyled",
                              "render": [Function],
                              "styledComponentId": "styles__HamburgerButtonStyled-sc-1iujtvx-2",
                              "target": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                              Symbol(Symbol.hasInstance): [Function],
                            }
                          }
                          forwardedRef={null}
                          onClick={[Function]}
                          type="button"
                        >
                          <BasicButton
                            aria-controls="navbarNav"
                            aria-expanded="false"
                            aria-label="Toggle navigation"
                            className="navbar-toggler "
                            data-target="#navbarNav"
                            data-toggle="collapse"
                            onClick={[Function]}
                            type="button"
                          >
                            <div
                              className="navbar-toggler "
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="button"
                              tabIndex={0}
                            >
                              <span
                                className="navbar-toggler-icon"
                                key=".0"
                              />
                            </div>
                          </BasicButton>
                        </StyledComponent>
                      </styles__HamburgerButtonStyled>
                      <div
                        className="collapse navbar-collapse"
                        id="navbarNav"
                      >
                        <ul
                          className="nav justify-content-end"
                        >
                          <li
                            className="nav-item"
                            key="AboutMe\\"_nav\\""
                          >
                            <styles__LinkStyled
                              activeClass="active"
                              duration={600}
                              offset={-70}
                              smooth={true}
                              to="ProfessionalIntro"
                            >
                              <StyledComponent
                                activeClass="active"
                                duration={600}
                                forwardedClass={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": undefined,
                                    "componentStyle": ComponentStyle {
                                      "componentId": "styles__LinkStyled-sc-1iujtvx-1",
                                      "isStatic": true,
                                      "lastClassName": "c1",
                                      "rules": Array [
                                        "padding:25px;text-transform:uppercase;cursor:pointer;border-bottom:solid 1px ",
                                        "#f6f6f6",
                                        ";&:hover{background-color:",
                                        "#f6f6f6",
                                        ";}",
                                      ],
                                    },
                                    "displayName": "styles__LinkStyled",
                                    "render": [Function],
                                    "styledComponentId": "styles__LinkStyled-sc-1iujtvx-1",
                                    "target": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                    Symbol(Symbol.hasInstance): [Function],
                                  }
                                }
                                forwardedRef={null}
                                offset={-70}
                                smooth={true}
                                to="ProfessionalIntro"
                              >
                                <Link
                                  activeClass="active"
                                  className="c1"
                                  duration={600}
                                  offset={-70}
                                  smooth={true}
                                  to="ProfessionalIntro"
                                >
                                  <LinkElement
                                    className="c1"
                                    onClick={[Function]}
                                  >
                                    <a
                                      className="c1"
                                      onClick={[Function]}
                                    >
                                      About Me
                                    </a>
                                  </LinkElement>
                                </Link>
                              </StyledComponent>
                            </styles__LinkStyled>
                          </li>
                          <li
                            className="nav-item"
                            key="Hobbies\\"_nav\\""
                          >
                            <styles__LinkStyled
                              activeClass="active"
                              duration={600}
                              offset={-70}
                              smooth={true}
                              to="Hobbies"
                            >
                              <StyledComponent
                                activeClass="active"
                                duration={600}
                                forwardedClass={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": undefined,
                                    "componentStyle": ComponentStyle {
                                      "componentId": "styles__LinkStyled-sc-1iujtvx-1",
                                      "isStatic": true,
                                      "lastClassName": "c1",
                                      "rules": Array [
                                        "padding:25px;text-transform:uppercase;cursor:pointer;border-bottom:solid 1px ",
                                        "#f6f6f6",
                                        ";&:hover{background-color:",
                                        "#f6f6f6",
                                        ";}",
                                      ],
                                    },
                                    "displayName": "styles__LinkStyled",
                                    "render": [Function],
                                    "styledComponentId": "styles__LinkStyled-sc-1iujtvx-1",
                                    "target": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                    Symbol(Symbol.hasInstance): [Function],
                                  }
                                }
                                forwardedRef={null}
                                offset={-70}
                                smooth={true}
                                to="Hobbies"
                              >
                                <Link
                                  activeClass="active"
                                  className="c1"
                                  duration={600}
                                  offset={-70}
                                  smooth={true}
                                  to="Hobbies"
                                >
                                  <LinkElement
                                    className="c1"
                                    onClick={[Function]}
                                  >
                                    <a
                                      className="c1"
                                      onClick={[Function]}
                                    >
                                      Hobbies
                                    </a>
                                  </LinkElement>
                                </Link>
                              </StyledComponent>
                            </styles__LinkStyled>
                          </li>
                          <li
                            className="nav-item"
                            key="ContactMe\\"_nav\\""
                          >
                            <styles__LinkStyled
                              activeClass="active"
                              duration={600}
                              offset={-70}
                              smooth={true}
                              to="ContactMePage"
                            >
                              <StyledComponent
                                activeClass="active"
                                duration={600}
                                forwardedClass={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": undefined,
                                    "componentStyle": ComponentStyle {
                                      "componentId": "styles__LinkStyled-sc-1iujtvx-1",
                                      "isStatic": true,
                                      "lastClassName": "c1",
                                      "rules": Array [
                                        "padding:25px;text-transform:uppercase;cursor:pointer;border-bottom:solid 1px ",
                                        "#f6f6f6",
                                        ";&:hover{background-color:",
                                        "#f6f6f6",
                                        ";}",
                                      ],
                                    },
                                    "displayName": "styles__LinkStyled",
                                    "render": [Function],
                                    "styledComponentId": "styles__LinkStyled-sc-1iujtvx-1",
                                    "target": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                    Symbol(Symbol.hasInstance): [Function],
                                  }
                                }
                                forwardedRef={null}
                                offset={-70}
                                smooth={true}
                                to="ContactMePage"
                              >
                                <Link
                                  activeClass="active"
                                  className="c1"
                                  duration={600}
                                  offset={-70}
                                  smooth={true}
                                  to="ContactMePage"
                                >
                                  <LinkElement
                                    className="c1"
                                    onClick={[Function]}
                                  >
                                    <a
                                      className="c1"
                                      onClick={[Function]}
                                    >
                                      ContactMe
                                    </a>
                                  </LinkElement>
                                </Link>
                              </StyledComponent>
                            </styles__LinkStyled>
                          </li>
                        </ul>
                      </div>
                    </nav>
                  </StyledComponent>
                </styles__Nav>
              </NavTopButtons>
              <Drawer
                isDrawerOpen={false}
                itemKeyAttr="key"
                itemLabelAttr="DisplayName"
                items={
                  Array [
                    Object {
                      "DisplayName": "About Me",
                      "LinkTo": "ProfessionalIntro",
                      "key": "AboutMe",
                    },
                    Object {
                      "DisplayName": "Hobbies",
                      "LinkTo": "Hobbies",
                      "key": "Hobbies",
                    },
                    Object {
                      "DisplayName": "ContactMe",
                      "LinkTo": "ContactMePage",
                      "key": "ContactMe",
                    },
                  ]
                }
                selectItem={[Function]}
              >
                <styles__DrawerStyled
                  isDrawerOpen={false}
                >
                  <StyledComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "styles__DrawerStyled-sc-2wdqmg-0",
                          "isStatic": false,
                          "lastClassName": "c2",
                          "rules": Array [
                            "position:fixed;top:0;left:",
                            [Function],
                            ";height:100%;width:300px;background:",
                            "#fff",
                            ";padding:0px;transition:0.25s ease-in-out;box-shadow:rgba(0,0,0,0.15) 0px 3px 10px,rgba(0,0,0,0.22) 0px 3px 10px;z-index:1000;color:",
                            "#000000",
                            ";&.drawerOpen{left:0;}",
                          ],
                        },
                        "displayName": "styles__DrawerStyled",
                        "render": [Function],
                        "styledComponentId": "styles__DrawerStyled-sc-2wdqmg-0",
                        "target": "div",
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    isDrawerOpen={false}
                  >
                    <div
                      className="c2"
                    >
                      <styles__LinkStyled
                        activeClass="active"
                        duration={600}
                        key="AboutMe"
                        offset={-70}
                        onClick={[Function]}
                        smooth={true}
                        to="ProfessionalIntro"
                      >
                        <StyledComponent
                          activeClass="active"
                          duration={600}
                          forwardedClass={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": undefined,
                              "componentStyle": ComponentStyle {
                                "componentId": "styles__LinkStyled-sc-2wdqmg-2",
                                "isStatic": true,
                                "lastClassName": "c3",
                                "rules": Array [
                                  "padding:25px;text-transform:uppercase;cursor:pointer;border-bottom:solid 1px ",
                                  "#f6f6f6",
                                  ";display:block;&:hover{background-color:",
                                  "#f6f6f6",
                                  ";}",
                                ],
                              },
                              "displayName": "styles__LinkStyled",
                              "render": [Function],
                              "styledComponentId": "styles__LinkStyled-sc-2wdqmg-2",
                              "target": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                              Symbol(Symbol.hasInstance): [Function],
                            }
                          }
                          forwardedRef={null}
                          offset={-70}
                          onClick={[Function]}
                          smooth={true}
                          to="ProfessionalIntro"
                        >
                          <Link
                            activeClass="active"
                            className="c3"
                            duration={600}
                            offset={-70}
                            onClick={[Function]}
                            smooth={true}
                            to="ProfessionalIntro"
                          >
                            <LinkElement
                              className="c3"
                              onClick={[Function]}
                            >
                              <a
                                className="c3"
                                onClick={[Function]}
                              >
                                About Me
                              </a>
                            </LinkElement>
                          </Link>
                        </StyledComponent>
                      </styles__LinkStyled>
                      <styles__LinkStyled
                        activeClass="active"
                        duration={600}
                        key="Hobbies"
                        offset={-70}
                        onClick={[Function]}
                        smooth={true}
                        to="Hobbies"
                      >
                        <StyledComponent
                          activeClass="active"
                          duration={600}
                          forwardedClass={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": undefined,
                              "componentStyle": ComponentStyle {
                                "componentId": "styles__LinkStyled-sc-2wdqmg-2",
                                "isStatic": true,
                                "lastClassName": "c3",
                                "rules": Array [
                                  "padding:25px;text-transform:uppercase;cursor:pointer;border-bottom:solid 1px ",
                                  "#f6f6f6",
                                  ";display:block;&:hover{background-color:",
                                  "#f6f6f6",
                                  ";}",
                                ],
                              },
                              "displayName": "styles__LinkStyled",
                              "render": [Function],
                              "styledComponentId": "styles__LinkStyled-sc-2wdqmg-2",
                              "target": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                              Symbol(Symbol.hasInstance): [Function],
                            }
                          }
                          forwardedRef={null}
                          offset={-70}
                          onClick={[Function]}
                          smooth={true}
                          to="Hobbies"
                        >
                          <Link
                            activeClass="active"
                            className="c3"
                            duration={600}
                            offset={-70}
                            onClick={[Function]}
                            smooth={true}
                            to="Hobbies"
                          >
                            <LinkElement
                              className="c3"
                              onClick={[Function]}
                            >
                              <a
                                className="c3"
                                onClick={[Function]}
                              >
                                Hobbies
                              </a>
                            </LinkElement>
                          </Link>
                        </StyledComponent>
                      </styles__LinkStyled>
                      <styles__LinkStyled
                        activeClass="active"
                        duration={600}
                        key="ContactMe"
                        offset={-70}
                        onClick={[Function]}
                        smooth={true}
                        to="ContactMePage"
                      >
                        <StyledComponent
                          activeClass="active"
                          duration={600}
                          forwardedClass={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": undefined,
                              "componentStyle": ComponentStyle {
                                "componentId": "styles__LinkStyled-sc-2wdqmg-2",
                                "isStatic": true,
                                "lastClassName": "c3",
                                "rules": Array [
                                  "padding:25px;text-transform:uppercase;cursor:pointer;border-bottom:solid 1px ",
                                  "#f6f6f6",
                                  ";display:block;&:hover{background-color:",
                                  "#f6f6f6",
                                  ";}",
                                ],
                              },
                              "displayName": "styles__LinkStyled",
                              "render": [Function],
                              "styledComponentId": "styles__LinkStyled-sc-2wdqmg-2",
                              "target": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                              Symbol(Symbol.hasInstance): [Function],
                            }
                          }
                          forwardedRef={null}
                          offset={-70}
                          onClick={[Function]}
                          smooth={true}
                          to="ContactMePage"
                        >
                          <Link
                            activeClass="active"
                            className="c3"
                            duration={600}
                            offset={-70}
                            onClick={[Function]}
                            smooth={true}
                            to="ContactMePage"
                          >
                            <LinkElement
                              className="c3"
                              onClick={[Function]}
                            >
                              <a
                                className="c3"
                                onClick={[Function]}
                              >
                                ContactMe
                              </a>
                            </LinkElement>
                          </Link>
                        </StyledComponent>
                      </styles__LinkStyled>
                    </div>
                  </StyledComponent>
                </styles__DrawerStyled>
              </Drawer>
            </NavigationBody>
          </div>
        </Navigation>
      </Connect(Navigation)>
    </withSaga(Connect(Navigation))>
  </withReducer(withSaga(Connect(Navigation)))>
</Provider>
`;
